addons/stock/models/stock_move.py

```diff
@@ -1775,8 +1775,12 @@ def _action_cancel(self):
                     move.move_dest_ids.filtered(lambda m: m.state != 'done')._action_cancel()
             else:
                 if all(state in ('done', 'cancel') for state in siblings_states):
-                    move.move_dest_ids.write({'procure_method': 'make_to_stock'})
-                    move.move_dest_ids.write({'move_orig_ids': [(3, move.id, 0)]})
+                    move_dest_ids = move.move_dest_ids
+                    move_dest_ids.write({
+                        'procure_method': 'make_to_stock',
+                        'move_orig_ids': [(3, move.id, 0)]
+                    })
+                    move_dest_ids._recompute_state()
```

Comment:
That part should be probably handle in write of stock.move. It's logical to recompute state when you remove the origin